### PR TITLE
meta: skip lighthouse CI when no vercel preview detected

### PR DESCRIPTION
**Pull request by avivkeller**
## Description

This changes the Lighthouse CI to skip when a Vercel preview is not detected

## Related Issues

#7253

### CODE REVIEW

**Comment by avivkeller** [comment]
The commit message didn't save :thinking:, regardless, the PR title will be used, right? If not, feel free to force push and fix it.

**Comment by avivkeller** [comment]
image
 Without a Vercel preview, the Lighthouse CI is skipped, as shown above.

File: .github/workflows/lighthouse.yml [hunk 1]
```diff
@@ -76,11 +93,11 @@ jobs:
...
-            ${{ steps.vercel_preview_url.outputs.url }}/en
-            ${{ steps.vercel_preview_url.outputs.url }}/en/about
-            ${{ steps.vercel_preview_url.outputs.url }}/en/about/previous-releases
-            ${{ steps.vercel_preview_url.outputs.url }}/en/download
-            ${{ steps.vercel_preview_url.outputs.url }}/en/blog
+            ${{ needs.get-vercel-preview.outputs.url }}/en
+            ${{ needs.get-vercel-preview.outputs.url }}/en/about
+            ${{ needs.get-vercel-preview.outputs.url }}/en/about/previous-releases
+            ${{ needs.get-vercel-preview.outputs.url }}/en/download
+            ${{ needs.get-vercel-preview.outputs.url }}/en/blog
```
Comment (AugustinMauroy) [inline .github/workflows/lighthouse.yml:100]: need to add learn here

Comment (avivkeller) [reply .github/workflows/lighthouse.yml:100]: Shouldn't that be a followup, given that this is specifically for skipping the workflow?

File: .github/workflows/lighthouse.yml [hunk 2]
```diff
@@ -27,12 +27,39 @@ permissions:
   pull-requests: write
 
 jobs:
-  lighthouse-ci:
+  get-vercel-preview:
     # We want to skip our lighthouse analysis on Dependabot PRs
```
Comment (bmuenzenmeyer) [inline .github/workflows/lighthouse.yml:31]: this comment needs to move with the ordering

File: .github/workflows/lighthouse.yml (hunk 1 above)
Comment (bmuenzenmeyer) [reply .github/workflows/lighthouse.yml:100]: this list was semi-intelligently designed not to be exhaustive of all pages, but to try to hit the major features

- homepage (global)
- markdown rendering (about)
- previous-releases (the release data / table)
- download (release data / os detection / lots of complexity)
- blog (blog data / rendering)

Comment (AugustinMauroy) [reply .github/workflows/lighthouse.yml:100]: Imo we should test the learn section to ensure everything is right. Yeah it's look like about but it's not same eg there are avatars in the learn page

File: .github/workflows/lighthouse.yml (hunk 2 above)
Comment (avivkeller) [reply .github/workflows/lighthouse.yml:31]: What do you mean? It's commenting on the line below (skipping the Dependabot).

Comment (bmuenzenmeyer) [reply .github/workflows/lighthouse.yml:31]: to me the comment nesting seems to suggest it is documenting the job directly below it, which is no longer lighthouse but get vercel preview.

it is probably just a small nit - sorry for noise

File: .github/workflows/lighthouse.yml (hunk 1 above)
Comment (ovflowd) [reply .github/workflows/lighthouse.yml:100]: Lighthouse is not about everything being right, but sampling some of it and measuring performance. Realistically speaking we don't want to sample everything nor people should take it too seriously.

Comment (bmuenzenmeyer) [reply .github/workflows/lighthouse.yml:100]: thanks @ovflowd - i've had this open all day struggling to find the words. 
if we ever get to that point, we should use playwright to make e2e tests. i suspect chromatic could do more too

**Review by bmuenzenmeyer — APPROVED** [review]

**Comment by avivkeller** [comment]
👋 follow-up https://github.com/nodejs/nodejs.org/pull/7361

**Pull request merged**